js/StickyScrollBarComponent.js

```diff
@@ -0,0 +1,138 @@
+
+    /**
+     * StickyScrollBarComponent
+     * @constructor
+     * @param {jQuery} $rootWindow - jQuery wrapper of the window object
+     * @param {jQuery} $html - jQuery wrapper of the html node
+     */
+    constructor ($rootWindow, $html) {
+        this.$elementWithStickyScrollBar = {};
+        this.$window = $rootWindow;
+        this.$html = $html;
+        this.$stickyScrollBar = $('<div class="sticky-scrollbar"><div class="sticky-scrollbar__inner"></div></div>');
+        this.$stickyScrollBarInner = this.$stickyScrollBar.children();
+        // cache methods with context
+        this.scrollStickyScrollBarWithContext = this.scrollStickyScrollBar.bind(this);
+        this.scrollElementWithStickyScrollBarByScrollBarWithContext = this.scrollElementWithStickyScrollBarByScrollBar.bind(this);
+        this.updateStickyScrollBarWithContext = this.updateStickyScrollBar.bind(this);
+    }
+
+    /**
+     * Initialise
+     * @param {jQuery} $element - jQuery object of the element that requires a sticky scroll bar
+     */
+    init ($element) {
+        if (typeof $element === 'undefined' || !$element) {
+            throw new Error('an element to add the scroller to must be passed to StickyScrollBarComponent');
+        }
+
+        if (typeof this.$window === 'undefined' || !this.$window) {
+            throw new Error('window must be passed to StickyScrollBarComponent');
+        }
+
+        if (typeof this.$html === 'undefined' || !this.$html) {
+            throw new Error('$html must be passed to StickyScrollBarComponent');
+        }
+
+        // Set $elementWithStickyScrollBar
+        this.$elementWithStickyScrollBar = $element;
+    
+        // Add the sticky scroll bar
+        this.$stickyScrollBar.appendTo(this.$elementWithStickyScrollBar);
+
+        // Init the sticky scroll bar
+        this.$stickyScrollBar
+            .addClass('u-display-none')
+            .on('scroll', this.scrollElementWithStickyScrollBarByScrollBarWithContext);
+
+        // Call on load
+        this.updateStickyScrollBarWithContext();
+
+        // Call on scroll and window resize
+        this.$window.on('scroll resize', this.updateStickyScrollBarWithContext);
+    }
+
+    /**
+     * Set $elementWithStickyScrollBar scroll when scrolled by sticky scroll bar
+     */
+    scrollElementWithStickyScrollBarByScrollBar () {
+        this.$elementWithStickyScrollBar.scrollLeft(this.$stickyScrollBar.scrollLeft());
+    }
+
+    /**
+     * Set $stickyScrollBar scroll when $elementWithStickyScrollBar is scrolled
+     */
+    scrollStickyScrollBar () {
+        this.$stickyScrollBar.scrollLeft(this.$elementWithStickyScrollBar.scrollLeft());
+    }
+
+    /**
+     * Toggle visiblity of sticky scroll bar
+     */
+    showStickyScrollBar (option) {
+        if (option) {
+            this.$stickyScrollBar.removeClass('u-display-none');
+        } else {
+            this.$stickyScrollBar.addClass('u-display-none');
+        }
```

Comment:
A little picky maybe – jQuery gives us an argument to `toggleClass` that could reduce this footprint somewhat
```javascript
showStickyScrollBar (option) {
    this.$stickyScrollBar.toggleClass('u-display-none', option);
}
```